experimental/dds/tree2/src/feature-libraries/editable-tree-2/proxies/proxies.ts

```diff
@@ -156,8 +157,11 @@ function createObjectProxy<TSchema extends ObjectNodeSchema>(
 							| OptionalField<AllowedTypes>;
 
 						const { content, hydrateProxies } = extractFactoryContent(value);
+						const offNextChange = editNode[onNextChange](() => {
+							hydrateProxies(typedField.boxedContent);
+						});
 						typedField.content = content;
-						hydrateProxies(typedField.boxedContent);
+						offNextChange();
```

Comment:
Even though the `onNextChange` event automatically deregisters itself after firing, we still have to manually deregister here. That's because `onNextChange` might actually never fire if, for example, the tree notices that we are inserting nothing of consequence (like inserting `[]` into a list, or setting `42` to `42`) and decides to early out. In that case, we need to unsubscribe manually here. Unsubscribing twice does no harm.

Reply:
Can we add this as a comment in the code here? Seems relevant for folks reading the code in the future.

Reply:
I see there are multiple places we need this. Maybe a wrapper function with the docs on it?

Reply:
Good suggestion, that really cleaned it up.